Issue #45 docker

ssilverm opened the pull request:
# Proposed Changes

This Dockerizes Quiz Extensions.

Resolves #45

ssilverm committed:
Dockerizing Quiz Extensions

ssilverm committed:
README

Thetwam committed:
Pull in latest develop; resolve conflicts

Thetwam reviewed (changes requested):
Need to add `FLASK_APP` and `FLASK_DEBUG` environment variables

Tests fail because the template for config.py is missing. Need to update `run-tests.yml` to support the new setup.

I noticed HTTPS isn't supported. I would like to retain that functionality.

ssilverm committed:
Updated README instructions.  Added FLASK variables to service

Thetwam commented:
Need to pull in latest `develop`, address conflicts, and address comments above.

ssilverm committed:
fixes, ssl, readme changes

ssilverm committed:
trying to fix github actions

ssilverm committed:
lint

ssilverm committed:
README lint

ssilverm committed:
env

ssilverm committed:
trying dotenv action

ssilverm committed:
testing actions

ssilverm committed:
trying dotenv to setenv

ssilverm committed:
seeing if this works...

ssilverm committed:
seeing if this works...

ssilverm committed:
Updating env template, readme, and tests workflow

ssilverm committed:
README lint

ssilverm committed:
switching from @v1 to @v2 codecov-action

ssilverm committed:
testing list files

ssilverm committed:
testing add xml

ssilverm commented on README.md:
```diff
@@ -14,6 +14,53 @@ all quizzes at once.
+python
+from views import db, app
+with app.app_context():
+    db.create_all()
```
Fixed.

ssilverm commented on docker-compose.production.yaml:
```diff
@@ -0,0 +1,48 @@
+#       - 6379:6379
+
+# volumes:
+#   quizext_db_data: {}
```
Removed these lines and added a comment about what is still needed in production.

ssilverm commented on config.py:
```diff
@@ -44,17 +44,9 @@
             "formatter": "standard",
             "class": "logging.StreamHandler",
         },
-        "file": {
-            "level": "INFO",
-            "formatter": "standard",
-            "class": "logging.handlers.RotatingFileHandler",
-            "filename": "logs/quiz_ext.log",
-            "maxBytes": 1024 * 1024 * 5,  # 5 MB
-            "backupCount": 5,
-        },
```
Because docker outputs everything to stdout which can then be piped wherever...

dgwn committed:
Address status page conflicts
- remove script_name from nginx.conf to fix index route
- map ports 80:80 and 443:443 in docker-compose.yml
- pin SQLAlchemy and Werkzeug versions
- use 'docker compose' commands instead of 'docker-compose'
- linting fixes
- gitignore docker's copy of ngingx.conf and uwsgi.ini

dgwn committed:
Github Actions and cleanup
- remove Python 3.7 from runner matrix
- use Python 3.8 in Docker image
- add newline to CONTRIBUTING.md
- ignore markdown-style.rb rule for 'ul-indent'
- use a single temporary Docker container for tests and tear down when complete
- update CHANGELOG.md

dgwn committed:
disable codecov for now

Thetwam commented:
Closing in favor of #70
